src/plugin/track/track.js

```diff
@@ -479,6 +480,7 @@ plugin.track.createAndAdd = function(options) {
   var trackNode = plugin.file.kml.ui.updatePlacemark({
     'feature': track
   });
+  ol.events.listen(track, goog.events.EventType.PROPERTYCHANGE, trackNode.onFeatureChange, trackNode);
```

Comment:
At a glance, the `updatePlacemark` call is creating a `plugin.file.kml.ui.KMLNode` which is already listening internally to this event in `plugin.file.kml.ui.KMLNode.prototype.setFeature`. That feature change handler seems like the place where you would want to listen for updates to the name property.

Reply:
Thanks. I was thinking of the `feature` in `plugin.file.kml.ui.KMLNode.prototype.setFeature` as being the underlying linestring, rather  than conceptually being the track. Will take another look.

On the second point, is there any guidance / conventions on which events get used where? Failing that, for a name or other track-level attribute, which should I use?

Reply:
The feature in a KML node is sort of a data wrapper around a geometry, style, and a bunch of other metadata and utility for drawing the geometry to the map and interacting with it. Generally speaking, features and geometries have a 1 to 1 mapping, though through the use of styles, you can have multiple geometries on a single feature. I don't think tracks ever do this, but they do use a special feature class called `DynamicFeature` which is how they update their geometries with changes in the timeline.

We generally have a convention of not using Openlayers' internal events whenever possible. They fire a lot of them, they add listeners for everything, and are generally unoptimized as a result. By default, Openlayers fires an event on every change to a property, which is a feature we disable because it's just unnecessary overhead that we don't care about 99% of the time. In lieu of that, we'll fire our own property change events for things we know we care about.

In this case:
1. Set the name on the `Feature` instance. Call `feature.dispatchEvent(new os.events.PropertyChangeEvent('label')`.